src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -261,6 +261,50 @@ Alias Visitor::visit(Parser::AliasWithoutBracketsContext* ctx) {
   return {visitExpressionPimpl(ctx->expression()), visit(ctx->var())};
 }
 
+// ____________________________________________________________________________________
+parsedQuery::BasicGraphPattern Visitor::toGraphPattern(
+    const ad_utility::sparql_types::Triples& triples) {
+  parsedQuery::BasicGraphPattern pattern{};
+  pattern._triples.reserve(triples.size());
+  for (const auto& triple : triples) {
+    auto toTripleComponent = []<typename T>(const T& item) {
+      namespace tc = ad_utility::triple_component;
+      if constexpr (ad_utility::isSimilar<T, Variable>) {
+        return TripleComponent{item};
+      } else if constexpr (ad_utility::isSimilar<T, Literal>) {
+        return TripleComponent{
+            tc::Literal::fromStringRepresentation(item.toSparql())};
+      } else if constexpr (ad_utility::isSimilar<T, Iri>) {
+        return TripleComponent{
+            tc::Iri::fromStringRepresentation(item.toSparql())};
+      } else if constexpr (ad_utility::isSimilar<T, BlankNode>) {
+        return TripleComponent{
+            ParsedQuery::blankNodeToInternalVariable(item.toSparql())};
+      } else {
+        static_assert(ad_utility::alwaysFalse<T>);
+      }
+    };
+    auto subject = std::visit(toTripleComponent, triple.at(0));
+    auto predicate = std::visit(
+        []<typename T>(const T& item) -> PropertyPath {
+          if constexpr (ad_utility::isSimilar<T, Variable>) {
+            return PropertyPath::fromVariable(item);
+          } else if constexpr (ad_utility::isSimilar<T, Iri>) {
+            return PropertyPath::fromIri(item.toSparql());
+          } else {
+            static_assert(ad_utility::isSimilar<T, Literal> ||
+                          ad_utility::isSimilar<T, BlankNode>);
+            AD_THROW("Literals or blank nodes are not valid predicates.");
+          }
+        },
+        triple.at(1));
```

Comment:
I would prefer having the two lambdas before the loop, then the loop becomes small and trivial.